core/src/main/java/com/datastax/oss/driver/api/core/metadata/NodeShardingInfo.java

```diff
@@ -0,0 +1,23 @@
+package com.datastax.oss.driver.api.core.metadata;
+
+import com.datastax.oss.driver.api.core.metadata.token.Token;
+
+/** Holds sharding information for a particular Node. */
+public interface NodeShardingInfo {
+
+  public int getShardsCount();
+
+  public String getPartitioner();
+
+  public String getShardingAlgorithm();
```

Comment:
Hmm, curiously enough, we don't actually use those values at all in the driver itself.

For sharding algorithm we don't actually look at it, since Scylla has only one algorithm (`biased-token-round-robin`). In the future, it makes sense that this is sent by Scylla if the algorithm gets ever changed.

However, partitioner (for example `org.apache.cassandra.partitioners.Murmur3Partitioner`) doesn't make sense - it's actually false that this is the partitioner that's used for shard calculation. When a CDC Partitioner is used, the sharding information is calculated with that partitioner, not with Murmur3 (Scylla still sends Murmur3 partitioner in that case).

I think that we can remove those getters from this interface - the user won't be calculating it on their own, only via `shardId()`. That way we don't have the confusing `getPartitioner` and `getShardingAlgorithm` is not that useful for the user since this class does the calculation for them.